### ISSUE TITLE
API - Context.Stash()

### ISSUE THREAD

**Issue by rogeralsing** [AsynkronIT/protoactor-dotnet#815]
A feature that came from the Akka heritage, is Stashing.
Stashing allows us to prepend a message back onto the mailbox.
So that it can be processed again.

In Akka you can explicitly stash whatever you want there. which in Proto.Actor you can only stash the current message back.
This is used in the EndpointWriter actor when the endpoint fails.
But. this could easily be handled in other ways. and I'm not sure this feature really justifies its own existence.

TLDR; Should we get rid of it?

There are other, much more elegant ways to handle this. e.g. there found be an injected stash object, much like how Behavior is implemented. and the actor code could then pull messages out of that stash whenever needed.

**Comment by rogeralsing** [comment]
From the Akka docs:

>Introduction
>Stashing enables an actor to temporarily buffer all or some messages that cannot or should not be handled using the actor’s current behavior.

>A typical example when this is useful is **if the actor has to load some initial state or initialize some resources before it can accept the first real message**. Another example is when the actor is **waiting for something to complete before processing the next message**.

One of the reasons why this exists was the blocking receive pipeline of Akka.
Proto.Actor does not have this. we could await some initial task on start. and then just process things that come in after this

**Comment by sudsy** [comment]
I use some actors as distributed workers and in that case I use stash to keep track of the last message (and context - so I know who to reply to) in case the worker fails and I need a retry.

My scenario could be handled by supervisor logic, but if that was the case the supervisor should ideally be able to replay the whole context to the failed actor rather than just the message.

**Comment by rogeralsing** [comment]
Keeping this API, closing
